open-company/open-company-web
# Refactor auth-link state strings to be Base64 encoded EDN maps

Trello card: https://trello.com/c/DWVGl7SV/2093-adjust-oauth-state-string-encoding-to-provide-greater-flexibility

Review with: https://github.com/open-company/open-company-auth/pull/86

This PR shouldn't affect any existing behavior, and does not add any user-noticeable behavior. It instead lays the groundwork for allowing clients to control the redirect location of OAuth flows.

To test:
- Onboard user with Slack
- [x] All clear?
- Onboard user with Google
- [x] All clear?
- Invite an email user
- [x] Could they successfully onboard?
- Add the Slack bot
- [x] All clear?
- Invite a Slack user
- [x] Could they successfully onboard?
- Share a post requiring authentication, open it in incognito browser, hit the login wall
- [x] Are you able to login from here?
- Try logging in using any other method that I haven't thought of...

Something that I recommend trying is adding a `:redirect-origin` key to the state maps of both Slack and Google here: https://github.com/open-company/open-company-web/blob/14856a4937f21cd1c3e0d6cb59d319443e80f5a5/src/oc/web/actions/user.cljs#L154-L167

So in the case of Slack, try:

```
{:team-id "open-company-auth"
 :redirect oc-urls/slack-lander-check
 :redirect-origin "https://google.com"}
```

Then try logging in with Slack and see if you are indeed redirected to google.
